Add release drafter.

mwz opened the pull request:
This PR adds [release drafter](https://github.com/marketplace/actions/release-drafter) config and a GitHub action to update the current release draft every time a PR gets merged to master.

I've set up the categories based on existing labels, but I also added
- `Maintenance` (I'd recommend adding a new label `chore` for any maintenance related work), and
- `Dependency updates` - for when scapegoat gets added to Scala Steward.

Closes #273.

mwz commented:
Did you get the release drafted correctly? I can't see the draft because I don't have the permissions, but I was wondering whether it worked. Might be also worth labelling all the PRs to get them categorised correctly.

mwz commented:
Hey @mccartney could you check if you can see the current release draft? I just want to check if it's working and it included all the PRs since the last release.
It would be a good idea to label all of those PRs before a new release so that they get categorised correctly in the release note.

mccartney commented:
I can see the draft the laundry list of fixes at https://github.com/sksamuel/scapegoat/releases/tag/untagged-d79439205a58a609cd68
I am surprised the list is so long. It's definitely too many items there. The oldest number included there is #48

mwz commented:
That doesn't sound right, it should only include the changes since v1.4.1. Maybe the ci job in travis doesn't fetch the tags on checkout, that's the first thing that comes to my mind.

mwz commented:
I'd try this and see if that makes any difference:
```
before_install:
  - git fetch --tags
```
